File: java/business-context/src/main/java/cz/filipklimes/diploma/framework/businessContext/expression/IsNotBlank.java
package cz.filipklimes.diploma.framework.businessContext.expression;

import cz.filipklimes.diploma.framework.businessContext.weaver.BusinessOperationContext;

import java.io.*;
import java.util.*;

public class IsNotBlank extends UnaryOperator<Boolean, String> implements Serializable
{

    public IsNotBlank(final Expression<String> argument)
    {
        super(argument);
    }

    @Override
    public Boolean interpret(final BusinessOperationContext context)
    {
        String string = getArgument().interpret(context);
        if (string == null) {
            return false;
        }
        return !string.isEmpty();
    }

    @Override
    public Map<String, String> getProperties()
    {
        return Collections.emptyMap();
    }

    @Override
    public String getName()
    {
        return "is-not-blank";
    }

    @Override
    public void accept(final ExpressionVisitor visitor)
    {
        visitor.visit(this);
    }

    @Override
    public String toString()
    {
        return String.format("%s is not blank", getArgument());
    }

}

File: nodejs/business-context/src/expression/IsNotBlank.js
export default class IsNotNull {
    constructor(argument) {
        this.argument = argument
    }

    interpret(context) {
        let string = this.argument.interpret(context)
        return string != null && typeof string === 'string' && string.length > 0
    }

    getName() {
        return 'is-not-blank'
    }

    getArguments() {
        return [this.argument]
    }

    getProperties() {
        return {}
    }

    toString() {
        return this.argument.toString() + " is not blank"
    }

    clone() {
        return new IsNotNull(this.argument.clone())
    }
}
